core/src/test/java/org/apache/brooklyn/core/entity/ApplicationLifecycleStateTest.java

```diff
@@ -201,6 +227,163 @@ public void testStartsThenChildFailsButWithQuorumCausesAppToStayHealthy() throws
         assertUpAndRunningEventually(app);
     }
 
+    /**
+     * Sensor value does not match the order of events. For example events are running, starting,
```

Comment:
Maybe reword the javadoc - I think this refers to the situation before you made the fix.

How about:
```
    /**
      * Tests concurrent modifications to a sensor, asserting that the last notification the subscribers 
      * receives equals the last value that sensor has.
      * 
      * Prior to this being fixed (see https://github.com/apache/brooklyn-server/pull/622), it caused 
      * problems in ComputeServiceIndicatorsFromChildrenAndMembers: it saw a child transition 
      * from "running" to "starting", and thus emitted the on-fire event for the parent entity. As asserted
      * by this test, the enricher should now always receive the events in the correct order (e.g. "starting",
      * "running").
      */
```